test/integration/encoder_timing.cc

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <gtest/gtest.h>
+
+#include "ignition/common/VideoEncoder.hh"
+#include "ignition/common/Video.hh"
+#include "test_config.h"
+#include "test/util.hh"
+
+using namespace ignition;
+using namespace common;
+
+const unsigned int kSize = 10;
+const unsigned char *kFrame = new unsigned char[kSize*kSize];
+
+// set to 720ms because video duration missing additional 18 frames
+//    which may be due to how video encoding works
+const std::chrono::milliseconds kTol(720);
+
+void durationTest(VideoEncoder &_vidEncoder, Video &_video,
+                  const int &_fps, const int &_seconds)
+{
+  _vidEncoder.Start("mp4", "", kSize, kSize, _fps, 0);
+
+  int frameCount = 0;
+  while (frameCount != _fps*_seconds)
+  {
+    if (_vidEncoder.AddFrame(kFrame, kSize, kSize))
+      ++frameCount;
+  }
+
+  _vidEncoder.Stop();
+  _video.Load(common::joinPaths(common::cwd(), "/TMP_RECORDING.mp4"));
+
+  EXPECT_NEAR(std::chrono::duration_cast<std::chrono::milliseconds>(
+              _video.Duration()).count(),
+              _seconds*1000,
+              kTol.count());
+}
+
+TEST(EncoderTimingTest, Duration)
+{
+  VideoEncoder vidEncoder;
+  Video video;
+
+  durationTest(vidEncoder, video, 50, 1);
+  durationTest(vidEncoder, video, 30, 2);
+  durationTest(vidEncoder, video, 25, 5);
+
+  delete [] kFrame;
```

Comment:
I think you either want to:

a) Allocate and deallocate this frame completely within the test, and pass it to duration test.
b) Allocate the deallocate the frame globally and don't do any memory management in the test.

Mixing the two could potentially make it easy to introduce bugs if more tests are ever added here.

Since this is also an array, you may consider using `std::array<unsigned char, kSize*kSize>`, or a `std::vector` to not have to deal with memory allocation at all.

Reply:
Thanks for the feedback. Better? 39ccc6ae3fef37c48c8d791a8d91ba1e03a291f6

Reply:
Looks great, just looks like you need to include `<array>` to make MacOS happy.

Reply:
done 67db98b0246542bc0dd6fcddcbae0121e5069bba. Is there a way to check things like this beforehand?

Reply:
Sometimes codecheck can catch it.  There are some explicit tools that will do it as well such as: https://include-what-you-use.org/

In this case it's generally good practice to do, but some compilers are a bit more picky about it.  There are many cases where an extra iteration or two may be spent appeasing Windows or Mac...